scripts/LSTM.py

```diff
@@ -1,315 +1,289 @@
-feature_names= []
-for feature_type in feature_types:
-    filepath = './scripts/feature_names/eq_week/' + feature_type + '.csv'
-    feature_type_name = pd.read_csv(filepath,header=None)
-    feature_type_name = feature_type_name.values.reshape(-1)
-    feature_names.append(feature_type_name)
-    print(feature_type_name.shape)
-
-if remove_obvious:
-  new_marras = feature_names[2][[2,3,4,5]]
-  feature_names[2] = new_marras
-
-  for course in courses:
-    new_features = feature_list[course][2].drop([0,1,6], axis=1)
-    feature_list[course][2] = new_features
-
-# reformat feature names
-# ex: time_sessions_<function sum at 0x7f3bd02cc9d0> -> time_sessions_sum
-def clean_name(feature):
-  id = feature.find('<')
-  if id==-1:
-    return feature
-  fct = feature[id+9:id+14].strip()
-  return feature[0:id]+fct
-
-feature_names = [np.array([clean_name(x) for x in feature_names[0]]),
- np.array([clean_name(x) for x in feature_names[1]]),
- np.array([clean_name(x) for x in feature_names[2]]),
- np.array([clean_name(x) for x in feature_names[3]])]
-
-for i,feature_type in enumerate(feature_types):
-    for course in courses:
-        y=np.isnan(feature_list[course][i]).sum(axis=0)/(feature_list[course][i].shape[0])
-        fig, ax = pyplot.subplots(figsize=(10, 5),facecolor='white')
-        g = sns.barplot(x=feature_names[i], y=y, palette="Greens", ax =ax)
-        g.set_xticklabels(g.get_xticklabels(),rotation=90)
-        g.set_title( feature_type+" for "+course, fontsize=15)
-        g.set_xlabel("", fontsize=15)
-        g.set_ylabel("% of NaNs", fontsize=12)
-        for index,data in enumerate(round(y,2)):
-            pyplot.text(x=index-.2 , y =data , s=f"{data}" , fontdict=dict(fontsize=10))
-        fig.savefig("NaNs_"+feature_type+"_for_"+course+".png", bbox_inches = 'tight', facecolor=fig.get_facecolor())
-        
-for i,feature_type in enumerate(feature_types):
-    for course in courses:
-        y=(feature_list[course][i]==0).sum(axis=0)/(feature_list[course][i].shape[0])
-        fig, ax = pyplot.subplots(figsize=(10, 5),facecolor='white')
-        g = sns.barplot(x=feature_names[i], y=y, palette="Greens",ax =ax)
-        g.set_xticklabels(g.get_xticklabels(),rotation=90)
-        g.set_title( feature_type+" for "+course, fontsize=15)
-        g.set_xlabel("", fontsize=15)
-        g.set_ylabel("% of zeros", fontsize=12)
-        for index,data in enumerate(round(y,2)):
-            pyplot.text(x=index-.2 , y =data , s=f"{data}" , fontdict=dict(fontsize=10))
-        fig.savefig("zeros_"+feature_type+"_for_"+course+".png", bbox_inches = 'tight', facecolor=fig.get_facecolor())
-
+    params[course] = {"num_epochs": 15}
 
 # ## Training on the dataset DSP001 with replacing NaNs:
 
 # Bidirection LSTM definition
 
-def bidirectional_lstm(x_train, y_train, x_test, y_test, x_val, y_val, week_type, feature_types, course,n_weeks,n_features, num_epochs=100):
-    n_dims = x_train.shape[0]
-    look_back = 3
+
+def bidirectional_lstm(
```

Comment:
I think it might be better to leave this as one line instead of expanding it (since it has so many params)